### ISSUE TITLE
rpython/tool/version.py will fail on git release bundles

### ISSUE THREAD

**Issue by mattip** [pypy/pypy#4855]
A repeat of issue #3315 where we released a source bundle without providing a git/hg hash file. There is a `git archive` command, but unlike `hg archive` it does not create a `hg-archival.txt` file. Another solution is needed so that `python rpython/tool/version.py` will print a `(branch, hash)` tuple.

**Comment by mattip** [comment]
I guess the easiest thing is to leave the name `.hg_archival.txt` and to recreate it by hand in the `repackage.sh` script.

**Comment by mattip** [comment]
The format of the file from mercurial is 
```
repo: 45eff22199974f85fd96138d25510f4660aba5a1
node: b29ea555846562061345eb58ca9bd717e1f0a9e3
branch: release-pypy3.10-v7.x
tag: release-pypy3.10-v7.3.14
```

I have all the information to do the node, branch, tag fields, which is more than enough to statisfy `rpython/tool/version.py`. The name is a bit weird, but maybe we can change that for a future release.

**Comment by mattip** [comment]
Fixed in #4858
